crates/typst/src/geom/gradient.rs

```diff
@@ -707,6 +725,26 @@ impl Gradient {
 }
 
 impl Gradient {
+    /// Clones this gradient, but with a different relative placement.
+    pub fn with_relative(&self, relative: Relative) -> Self {
```

Comment:
This could take `self` and then use `Arc::make_mut` to prevent cloning if the `Arc` is already unique.

Reply:
Well in this case it never is, but I did the change ;)